tracer/src/Datadog.Trace/AppSec/InstrumentationGateway.cs

```diff
@@ -111,7 +117,8 @@ private void RaiseEvent(HttpContext context, Span relatedSpan, Func<IDictionary<
 
         internal void RaiseBlockingOpportunity(HttpContext context, Scope scope, ImmutableTracerSettings tracerSettings, Action<InstrumentationGatewayBlockingEventArgs> doBeforeActualBlocking = null)
         {
-            BlockingOpportunity?.Invoke(this, new InstrumentationGatewayBlockingEventArgs(context, scope, tracerSettings, doBeforeActualBlocking));
+            var ea = new InstrumentationGatewayBlockingEventArgs(context, scope, tracerSettings, doBeforeActualBlocking);
+            BlockingOpportunity?.Invoke(this, ea);
```

Comment:
This now creates the `InstrumentationGatewayBlockingEventArgs` object even if `BlockingOpportunity` is null, that looks worse for allocations to me? 🤔

Reply:
Agree, fixed in 52d5decfa